lib/api/controllers/securityController.js

```diff
@@ -175,4 +182,180 @@ module.exports = function SecurityController (kuzzle) {
+      .then(user => {
+        var response;
+
+        if (!user) {
+          return Promise.resolve(new ResponseObject(requestObject, {found: false}));
+        }
+
+        if (requestObject.data.body.hydrate !== undefined && requestObject.data.body.hydrate === false) {
+          response = {
+            _id: user._id,
+            _source: kuzzle.repositories.user.serializeToDatabase(user)
+          };
+        }
+        else {
+          response = {
+            _id: user._id,
+            _source: user
+          };
+        }
+        delete response._source._id;
+
+        return Promise.resolve(new ResponseObject(requestObject, response));
+      });
+  };
+
+  /**
+   * Returns the user identified by the given jwt token
+   * @param {RequestObject} requestObject
+   * @param context
+   * @returns {Promise}
+   */
+  this.getCurrentUser = function (requestObject, context) {
+    requestObject.data._id = context.user._id;
+
+    return this.getUser(requestObject);
+  };
+
+  /**
+   * Returns the User objects matching the given filters
+   * @param {RequestObject} requestObject
+   * @returns {Promise}
+   */
+  this.searchUsers = function (requestObject) {
+    var
+      filter = {},
+      from = 0,
+      size = 20,
+      hydrate = true;
+
+    kuzzle.pluginsManager.trigger('security:searchUsers', requestObject);
+
+    if (requestObject.data.body.filter !== undefined) {
+      filter = requestObject.data.body.filter;
+    }
+    if (requestObject.data.body.from !== undefined) {
+      from = requestObject.data.body.from;
+    }
+    if (requestObject.data.body.size !== undefined) {
+      size = requestObject.data.body.size;
+    }
+    if (requestObject.data.body.hydrate !== undefined) {
+      hydrate = requestObject.data.body.hydrate;
+    }
+
+    if (!hydrate) {
+      return kuzzle.repositories.user.search(filter, from, size, hydrate);
+    }
+
+    return kuzzle.repositories.user.search(filter, from, size, hydrate)
+      .then(responseObject => {
+        responseObject.data.body.hits = responseObject.data.body.hits.map(user => {
+          var doc = {
+            _id: user._id,
+            _source: user
+          };
+          delete doc._source._id;
+
+          return doc;
+        });
+
+        return Promise.resolve(responseObject);
+      });
+  };
+
+  /**
+   * Deletes a User from Kuzzle
+   * @param {RequestObject} requestObject
+   * @returns {Promise}
+   */
+  this.deleteUser = function (requestObject) {
+    kuzzle.pluginsManager.trigger('security:deleteUser', requestObject);
+
+    if (!requestObject.data._id) {
+      return Promise.reject(new BadRequestError('No user id given'));
+    }
+
+    return kuzzle.repositories.user.delete(requestObject.data._id)
+      .then(() => {
+        return new ResponseObject(requestObject, {});
+      });
+  };
+
+  this.createUser = function (requestObject) {
+    var user = new User();
+
+    kuzzle.pluginsManager.trigger('data:createUser', requestObject);
```

Comment:
We should use pipe event here because of strategy: each plugin strategy may insert custom data into user (add an hashed version of password in case of local strategy for instance)